backend/python-functions/data_ingest.py

```diff
@@ -0,0 +1,145 @@
+        'staffDeaths': row['Staff Deaths'],
+        'staffTested': row['Staff Tested'],
+        'staffTestedNegative': row['Staff Tested Negative'],
+        'staffTestedPositive': row['Staff Tested Positive'],        
+    }
+    
+    # Remove empty strings and convert non-empty strings to integers
+    return {k: int(v) for k, v in covid_case_counts.items() if v}
+
+def download_covid_case_data(bucket_name, file_name):
+    storage_client = storage.Client()
+    blob = storage_client.get_bucket(bucket_name).get_blob(file_name) 
+
+    download_location = f'/tmp/{file_name}'
+
+    blob.download_to_filename(download_location)    
+
+    return download_location
+
+def reshape_facilities_data(file_location):
+    """
+    This function reshapes the daily Covid case data provided in CSV format into a 
+    dictionary structure that: 1) Removes repetitive and unwanted data and 2) Aligns
+    with how we ultimately persist our documents within Firestore.
+
+    For example, the following CSV:
+
+    Date,Facility Type,State,Canonical Facility Name,Pop Tested,Pop Tested Positive, etc.
+    2020-06-01,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+    2020-06-02,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+    2020-06-03,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+    2020-06-04,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+
+    Would be reshaped to:
+
+    {
+      "Colorado::Denver Reception & Diagnostic Center": {
+        "canonicalName": "Denver Reception & Diagnostic Center",
+        "facilityType": "State Prisons",
+        "state": "Colorado",
+        "covidCases": {
+          "2020-06-01": {
+            "popDeaths": 0,
+            "popTested": 253,
+            "popTestedPositive": 2
+          },
+          "2020-06-02": {
+            "popDeaths": 0,
+            "popTested": 253,
+            "popTestedPositive": 2
+          },
+          "2020-06-03": {
+            "popDeaths": 0,
+            "popTested": 253,
+            "popTestedPositive": 2
+          },
+          "2020-06-04": {
+            "popDeaths": 0,
+            "popTested": 253,
+            "popTestedPositive": 2
+          }
+        }
+      },
+      ...
+    }
+    """
+    facilities = {}    
+    with open(file_location) as csv_file:
+        reader = csv.DictReader(csv_file, delimiter=',')
+
+        for row in reader:
+            state = row["State"].strip()
+            canonical_facility_name = row["Canonical Facility Name"].strip()
+            facility_type = row['Facility Type'].strip()
+            key = f'{state}::{canonical_facility_name}'
+            
+            if(key in facilities):
+                # If the facility already exists, append the case counts for a given day.
+                facilities[key]['covidCases'][row['Date']] = build_covid_case_counts(row)
```

Comment:
I would pull up the indexing on date to the top of the loop alongside `state` and `facility_type`, e.g. `date = row['Date']`. Then just reference that in both places in this function. It will be easier to read in both places it's being used right now, in my opinion.

Reply:
I dig it 👍